ilastik/workflows/carving/carvingGui.py

```diff
@@ -252,11 +252,17 @@ def makeColortable():
             self._doneSegmentationColortable.append(QColor(0,255,0).rgba())
         makeColortable()
         self._updateGui()
-    
+
+    def _is_3d(self):
+        tagged_shape = defaultdict(lambda: 1)
+        tagged_shape.update(self.topLevelOperatorView.InputData.meta.getTaggedShape())
+        is_3d = tagged_shape['x'] > 1 and tagged_shape['y'] > 1 and tagged_shape['z'] > 1
+        return is_3d
+
     def _after_init(self):
         super(CarvingGui, self)._after_init()
-        if self.render:self._toggleSegmentation3D()
-        
+        if self.render:
+            self._toggleSegmentation3D()
```

Comment:
! really nice one. Didn't understand the old line at first glance :D

Reply:
😆